app/controllers/users/companies_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Users
+  class CompaniesController < BaseController
+    expose :company
+
+    def new; end
+
+    def create
+      self.company = CreateCompany.call(company_params: company_params, user: current_user).company
+
+      respond_with company, location: root_path
+    end
```

Comment:
It makes sense, but errors on form more useful than "Company can't be created". I don't like this extra information. In this case error from context is redundant at all, I guess. However, I will leave them for a possible bright future. :)

Reply:
Could you please clarify "but errors on form more useful than "Company can't be created". We do `respond_with company` in two options. This means that errors on the form will be shown in both cases.

Reply:
I want to say that the interactor error does not carry any additional information for the user. According to the errors in the form, he already sees that the company has not been created.

Reply:
But in this case, the user can not understand why a company can not be created if the owner can not be created.

Reply:
Do you mean that the form will be green, but the company will not be created? And in this case it would be necessary to let the user know that something went wrong? So that he could at least write to support, for example?

Reply:
Yes. But if you don't think it's necessary, you can leave it as it is.

Reply:
I didn't take this case into account. In that case, it's hard to disagree with you. Does it make sense then to show the interactor error only when the company has no errors? Will it be more correct or is it an overcomplication?

Reply:
Yes, you can do it.

Reply:
But the code will probably look complicated.